doc/api/esm.md

```diff
@@ -1053,11 +1053,22 @@ and parent URL. The module specifier is the string in an `import` statement or
 `import()` expression, and the parent URL is the URL of the module that imported
 this one, or `undefined` if this is the main entry point for the application.
 
+The `conditions` property on the `context` is an array of conditions for
+[Conditional Exports][] that apply to this resolution request. They can be used
+for looking up conditional mappings elsewhere or to modify the list when calling
+the default resolution logic.
+
+The [current set of Node.js default conditions][Conditional Exports] will always
+be in the `context.conditions` list passed to the hook. If the hook wants to
+ensure Node.js-compatible resolution logic, all items from this default
+condition list **must** be passed through to the `defaultResolve` function.
+
 ```js
 /**
  * @param {string} specifier
  * @param {object} context
  * @param {string} context.parentURL
+ * @param {string[]} context.conditions
```

Comment:
I wish this line read the following.

```javascript
@param {Array<string>} context.conditions
```

That would make this annotation compatible w/ Closure.

Reply:
I think for practical purposes I'd rather use syntax familiar to Typescript devs since *generally* I would expect the number of people using VSCode and/or TypeScript to be a lot bigger than the number of people actively using Closure Compiler.